neuralmonkey/runners/xent_runner.py

```diff
@@ -9,25 +9,26 @@
 from neuralmonkey.runners.base_runner import BaseRunner
 
 
-class PerplexityRunner(BaseRunner[AutoregressiveDecoder]):
+SupportedDecoders = Union[AutoregressiveDecoder]
+
+
+class XentRunner(BaseRunner[SupportedDecoders]):
 
     # pylint: disable=too-few-public-methods
     # Pylint issue here: https://github.com/PyCQA/pylint/issues/2607
-    class Executable(BaseRunner.Executable["PerplexityRunner"]):
+    class Executable(BaseRunner.Executable["XentRunner"]):
 
         def collect_results(self, results: List[Dict]) -> None:
-            perplexities = np.mean(
-                [2 ** res["xents"] for res in results], axis=0)
-            xent = float(np.mean([res["xents"] for res in results]))
-            self.set_runner_result(outputs=perplexities.tolist(),
-                                   losses=[xent])
+            xents = np.concatenate([res["xents"] for res in results], axis=0)
```

Comment:
tohle nechceš konkatenovat ale průměrovat, nebo vynucovat, aby `len(results)` bylo 1. To jsou ensembly.